pkgs/development/python-modules/rustworkx/default.nix

```
@@ -0,0 +1,47 @@
+{ fetchFromGitHub
+, buildPythonPackage
+, rustPlatform
+, setuptools-rust
+, numpy
+, fixtures
+, networkx
+, libiconv
+, stdenv
+, lib
+}:
+
+buildPythonPackage rec {
+  pname = "rustworkx";
+  version = "0.12.1";
+
+  src = fetchFromGitHub {
+    owner = "Qiskit";
+    repo = pname;
+    rev = version;
+    hash = "sha256-d/KCxhJdyzhTjwJZ+GsXJE4ww30iPaXcPngpCi4hBZw=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    hash = "sha256-imhiPj763iumRQb+oeBOpICD1nCvzZx+3yQWu1QRRQQ=";
+  };
+
+  nativeBuildInputs = [ setuptools-rust ] ++ (with rustPlatform; [
+    cargoSetupHook
+    rust.cargo
+    rust.rustc
+  ]);
+
+  buildInputs = [ numpy ] ++ lib.optionals stdenv.isDarwin [ libiconv ];
```

```suggestion
  buildInputs = [ numpy libiconv ];
```
FYI libiconv is null on linux but does not matter to much

---

I guess it's better this is explicit

---

For rust programs this does not really matter to much for C/C++ iconv also exists for bsd systems, so it should just be used as is.